src/components/change-wallet/WalletList.tsx

```diff
@@ -80,18 +85,71 @@ const WalletListDivider = styled(Divider).attrs(({ theme: { colors } }: any) =>
   marginTop: -1,
 });
 
+const EditButton = styled(ButtonPressAnimation).attrs(({ editMode }: { editMode: boolean }) => ({
+  scaleTo: 0.96,
+  wrapperStyle: {
+    width: editMode ? 70 : 58,
+  },
+  width: editMode ? 100 : 100,
+}))(
+  IS_IOS
+    ? {
+        position: 'absolute',
+        right: 20,
+        top: -11,
+      }
+    : {
+        elevation: 10,
+        position: 'relative',
+        right: 20,
+        top: 6,
+      }
+);
+
+const EditButtonLabel = styled(Text).attrs(({ theme: { colors }, editMode }: { theme: any; editMode: boolean }) => ({
+  align: 'right',
+  color: colors.appleBlue,
+  letterSpacing: 'roundedMedium',
+  size: 'large',
+  weight: editMode ? 'bold' : 'semibold',
+  numberOfLines: 1,
+  ellipsizeMode: 'tail',
+}))({
+  height: 40,
+});
+
+const FOOTER_HEIGHT = getExperimetalFlag(HARDWARE_WALLETS) ? 100 : 60;
+const LIST_PADDING_BOTTOM = 6;
+export const MAX_LIST_HEIGHT = DEVICE_HEIGHT - 220;
+const WALLET_ROW_HEIGHT = 59;
+const WATCH_ONLY_BOTTOM_PADDING = IS_ANDROID ? 20 : 0;
+
+const getWalletListHeight = (numWallets: number, watchOnly: boolean) => {
+  let listHeight = !watchOnly ? FOOTER_HEIGHT + LIST_PADDING_BOTTOM : WATCH_ONLY_BOTTOM_PADDING;
+
+  if (numWallets) {
+    for (let i = 0; i < numWallets; i++) {
+      listHeight += WALLET_ROW_HEIGHT + 6;
+
+      if (listHeight > MAX_LIST_HEIGHT) {
+        return MAX_LIST_HEIGHT;
+      }
+    }
```

Comment:
do we need a loop? `Math.max(MAX_LIST_HEIGHT, (WALLET_ROW_HEIGHT + 6) * numWallets)`